File: desafio-02/MatMercer/c/primos.c
#include <stdio.h>

#define RANGE 10000

int isPrimo(int num);

int main() {

    int count = 1;
    while (count < 3) {
        printf("%d\n", ++count);
    }

    while(count <= RANGE) {
        count += 2;

        if (isPrimo(count)) {
            printf("%d\n", count);
        }
    }

    return 0;
}

// espera um numero positivo impar maior que zero, podemos ignorar check para números pares
int isPrimo(int numImpar) {
    if (numImpar % 3 == 0) {
        return 0;
    }

    // 6k +- 1 check
    // podemos parar em raiz de N
    for (int k = 5; k * k <= numImpar; k += 6) {
        if (
                numImpar % k == 0 // 6k - 1 check
                || numImpar % (k + 2) == 0 // 6k + 1 check
           ) {
            return 0;
        }
    }

    return 1;
}

File: desafio-02/Vinissokkj/python/primos.py
def eratosthenes(num):
    primo = [True for _ in range(num+1)]
    p = 2
    while p*p<=num:
        if primo[p]:
            for i in range(p*p,num+1,p):
                primo[i] = False
        p+=1
    for p in range(2,num+1):
        if primo[p]:
            print(p)

eratosthenes(10000)